components/Forms/RegisterForm/Utils.tsx

```diff
@@ -0,0 +1,50 @@
+export interface IFormErrors {
+    emailErrorMessage?: string;
+    usernameErrorMessage?: string;
+    passwordErrorMessage?: string;
+}
+
+export const emptyFormErrors = (): IFormErrors => {
+    return {
+        emailErrorMessage: '',
+        usernameErrorMessage: '',
+        passwordErrorMessage: ''
+    }
+}
+
+export const validateForm = (email: string, userName: string, password: string) : {isValid: boolean, formErrors: IFormErrors} => {
+    const {emailIsValid, emailErrorMessage} = validateEmail(email);
+    const {usernameIsValid, usernameErrorMessage} = validateUserName(userName);
+    const {passwordIsValid, passwordErrorMessage} = validatePassword(password);
+
+    if (emailIsValid && usernameIsValid && passwordIsValid) {
+        return {isValid: true, formErrors: emptyFormErrors()}
+    } 
+    return {isValid: false, formErrors: {emailErrorMessage: emailErrorMessage, usernameErrorMessage: usernameErrorMessage, passwordErrorMessage: passwordErrorMessage}}
+}
+
+const validateEmail = (email: string) : {emailIsValid: boolean, emailErrorMessage: string} => {
+    var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
vad betyder re? Är det här tecken som inte får vara med i email?

Reply:
Ja men precis, det här är ett rätt standard _regular expression_ för att kolla om en sträng passar in på formatet av en mailadress. Den säger liksom vad som får och inte får förekomma på olika platser i strängen